src/main/java/emissary/core/IBaseDataObjectXmlCodecs.java

```diff
@@ -0,0 +1,688 @@
+    /**
+     * The Base64 decoder.
+     */
+    private static final Base64.Decoder BASE64_DECODER = Base64.getMimeDecoder();
+    /**
+     * The XML attribute name for Encoding.
+     */
+    public static final String ENCODING_ATTRIBUTE_NAME = "encoding";
+    /**
+     * The XML element name for Name.
+     */
+    public static final String NAME_ELEMENT_NAME = "name";
+    /**
+     * A map of element names of IBaseDataObject methods that get/set primitives and their default values.
+     */
+    public static final Map<String, Object> PRIMITVE_NAME_DEFAULT_MAP = Collections
+            .unmodifiableMap(new ConcurrentHashMap<>(Stream.of(
+                    new SimpleEntry<>(BIRTH_ORDER_ELEMENT_NAME, new BaseDataObject().getBirthOrder()),
+                    new SimpleEntry<>(BROKEN_ELEMENT_NAME, new BaseDataObject().isBroken()),
+                    new SimpleEntry<>(NUM_CHILDREN_ELEMENT_NAME, new BaseDataObject().getNumChildren()),
+                    new SimpleEntry<>(NUM_SIBLINGS_ELEMENT_NAME, new BaseDataObject().getNumSiblings()),
+                    new SimpleEntry<>(OUTPUTABLE_ELEMENT_NAME, new BaseDataObject().isOutputable()),
+                    new SimpleEntry<>(PRIORITY_ELEMENT_NAME, new BaseDataObject().getPriority()))
+                    .collect(Collectors.toMap(SimpleEntry::getKey, SimpleEntry::getValue))));
+    /**
+     * The XML namespace for "xml".
+     */
+    public static final Namespace XML_NAMESPACE = Namespace.getNamespace(XMLConstants.XML_NS_PREFIX,
+            XMLConstants.XML_NS_URI);
+
+    /**
+     * Interface for decoding an element value.
+     */
+    public interface ElementDecoder {
+        /**
+         * Decodes and XML element value and sets it on the specified IBDO method.
+         * 
+         * @param elements the list of elements to be decoded.
+         * @param ibdo the ibdo to set the values on.
+         * @param ibdoMethodName the ibdo method name to use to set the values.
+         * @throws Exception thrown if anything goes wrong.
+         */
+        void decode(List<Element> elements, IBaseDataObject ibdo, String ibdoMethodName) throws Exception;
+    }
+
+    /**
+     * Interface for encoding an element value.
+     */
+    public interface ElementEncoder<T> {
+        /**
+         * Encodes a list of values into an element that is attached to the parent element with the specified child element
+         * name.
+         * 
+         * @param values to be encoded.
+         * @param parentElement that the child element is to be attached to.
+         * @param childElementName the name of the child element.
+         */
+        void encode(List<T> values, Element parentElement, String childElementName);
+    }
+
+    /**
+     * Class that contains the element decoders.
+     */
+    public static class ElementDecoders {
+        /**
+         * Decoder for boolean elements.
+         */
+        public final ElementDecoder booleanDecoder;
```

Comment:
Could be a follow-on PR, but it would be nice to make the `ElementDecoder` instances private and leverage them through type-specific methods - i.e., 
```java
   public void decodeBoolean(List<Element> elements, IBaseDataObject ibdo, String methodName) throws Exception {
       booleanDecoder.decode(elements, ibdo, methodName);
   }
```

That makes the calling pattern a bit cleaner.  For an even cleaner API, ElementDecoders could leverage a static `Map<String, String>` of ElementName -> methodName, so the API would look like:
```java
   public void decodeBoolean(Element currentElement, IBaseDataObject ibdo, String elementName) throws Exception {
       List<Element> elements = currentElement.getChildren(elementName)
       String methodName = methodMap.get(elementName);
       Validate.notNull(methodName, "Could not find the IBDO method for element name " + elementName);
       booleanDecoder.decode(elements, ibdo, methodName);
   }
```
That would simplify the usage in IBaseDataObjectXmlHelper from:
```java
decoders.booleanDecoder.decode(element.getChildren(OUTPUTABLE_ELEMENT_NAME), ibdo, OUTPUTABLE_SET_METHOD_NAME);
```
to 
```java
decoders.decodeBoolean(element, ibdo, OUTPUTABLE_ELEMENT_NAME);
```